File: java/java-practise/src/main/java/day/six/java-cucumber-healthcare/src/test/java/bdd/steps/LoginSteps.java
package bdd.steps;

import domain.AuthenticationService;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

import static org.assertj.core.api.Assertions.assertThat;

public class LoginSteps {

    private AuthenticationService authenticationService;
    private boolean loginSuccessful;

    @Given("the authentication system contains username {string} with password {string}")
    public void theAuthenticationSystemContainsUser(
            String username,
            String password
    ) {
        authenticationService = new AuthenticationService();
        authenticationService.addUser(username, password);
    }

    @When(
            "I log in with username {string} and password {string}"
    )
    public void iLogIn(
            String username,
            String password
    ) {
        loginSuccessful =
                authenticationService.login(username, password);
    }

    @Then("the login result should be {string}")
    public void theLoginResultShouldBe(String expectedResult) {
        boolean expectedSuccessful =
                switch (expectedResult.toLowerCase()) {
                    case "successful" -> true;
                    case "rejected" -> false;

                    default -> throw new IllegalArgumentException(
                            "Unsupported login result: "
                                    + expectedResult
                    );
                };

        assertThat(loginSuccessful)
                .isEqualTo(expectedSuccessful);
    }
}

File: scharp/day/six/CSharpSpecFlowHealthcareExample/Bindings/LoginSteps.cs
using CSharpSpecFlowHealthcareExample.Domain;
using NUnit.Framework;
using TechTalk.SpecFlow;

namespace CSharpSpecFlowHealthcareExample.Bindings
{
    [Binding]
    public sealed class LoginSteps
    {
        private AuthenticationService _authenticationService = null!;
        private bool _loginSuccessful;

        [Given(
            @"^the authentication system contains username ""([^""]*)"" " +
            @"with password ""([^""]*)""$")]
        public void GivenTheAuthenticationSystemContainsUser(
            string username,
            string password)
        {
            _authenticationService =
                new AuthenticationService();

            _authenticationService.AddUser(
                username,
                password);
        }

        [When(
            @"^I log in with username ""([^""]*)"" " +
            @"and password ""([^""]*)""$")]
        public void WhenILogIn(
            string username,
            string password)
        {
            _loginSuccessful =
                _authenticationService.Login(
                    username,
                    password);
        }

        [Then(
            @"^the login result should be ""([^""]*)""$")]
        public void ThenTheLoginResultShouldBe(
            string expectedResult)
        {
            var expectedSuccessful =
                expectedResult switch
                {
                    "successful" => true,
                    "rejected" => false,

                    _ => throw new ArgumentException(
                        $"Unknown result: {expectedResult}")
                };

            Assert.That(
                _loginSuccessful,
                Is.EqualTo(expectedSuccessful));
        }
    }
}
